File: src/main/java/br/ufu/facom/petsi/controlaPET/model/Loan.java
package br.ufu.facom.petsi.controlaPET.model;

import br.ufu.facom.petsi.controlaPET.model.enums.LoanStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "loans")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Loan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;

    @Column(nullable = false)
    int quantity;

    @Column(nullable = false)
    private LocalDate checkoutDate;

    @Column(nullable = false)
    private LocalDate expectedReturnDate;

    private LocalDate actualReturnDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LoanStatus status;
}

File: frontend/src/interfaces/Loan.ts
export type LoanStatus = 'ACTIVE' | 'COMPLETED' | 'OVERDUE'

export interface Loan {
    id: number,
    userName: string,
    itemId: number,
    itemName: string,
    quantity: number,
    checkoutDate: string,
    expectedReturnDate: string,
    actualReturnDate: string,
    status: LoanStatus
}

export interface CreateLoanRequest {
    itemId: number,
    quantity: number,
    expectedReturnDate: string
}